### ISSUE TITLE
Update Readme file with requirements section

### ISSUE THREAD

**Issue by nmarcetic** [absmach/magistrala#899]
**FEATURE REQUEST**

1. Is there an open issue addressing this request? If it does, please add a "+1" reaction to the
   existing issue, otherwise proceed to step 2.

2. Describe the feature you are requesting, as well as the possible use case(s) for it.
  Update Readme file with the `Requirements` section just before `Install`,  which will list linked Mainflux dependencies and build tools required for the running platform but build also (for developers). 
* Docker engine version - We use template 3.7
* docker-compose version 
* Go version
* protoc version - In order to build .proto files 

Add this info to [doc also](https://mainflux.readthedocs.io/en/latest/getting-started/) 

3. Indicate the importance of this feature to you (must-have, should-have, nice-to-have).
 must have

**Comment by nwneisen** [comment]
:laughing: I think I ran into an issue with 3/4 of these in the last week.  My current versions are:

docker-compose: 1.24.1
go: 1.13.3
protoc: 3.6.1

It might also be useful to point out that the current versions for [go](https://github.com/mainflux/mainflux/blob/master/scripts/ci.sh#L8) and [protoc](https://github.com/mainflux/mainflux/blob/master/scripts/ci.sh#L32) will be in the CI script.  This would be just in case the documentation fell behind.

I'd be happy to put in this change.  It might be a good time to also address 2 other issues I had getting started that I've been meaning to create an issue for.

1. Building forked repo - When I try to `go get` or `git clone` and `make` my forked repo, I get ~12 errors similar to the following:
`cmd/users/main.go:173:2: cannot use db (type *"github.com/mainflux/mainflux/vendor/github.com/jmoiron/sqlx".DB) as type *"github.com/nwneisen/mainflux/vendor/github.com/jmoiron/sqlx".DB in return argument`   
 I was able to get around these errors by cloning into a specific folder that matched the original mainflux repo's file path:
`git clone <my forked repo> github.com/mainflux/mainflux`    
 Rereading the [Developer's Guide](https://mainflux.readthedocs.io/en/latest/dev-guide/), I see that it has the user `go get` from the original mainflux repo.  I believe forking is typically expected for working on a project and branching is usually blocked on the original repo, except for maintainers.

2.  Suggested development workflow - There's a lot of information in the Developer's Guide.  I think it would be good to provide a basic workflow for new developers.  I looked at the `make rundev` command for a while before figuring out `make && make dockers_dev && make run` was more what I was looking for.  I would almost suggest if the Developer's Guide could be broken into a "Basic Developers Guide" and "Advanced Developers Guide" then this would help new developers get their feet wet.

These were the ways I found to get started.  There maybe be more preferred methods that I'm not aware of.

**Comment by drasko** [comment]
@nwneisen having this in mind, I think that you are the best candidate for this issue :).

Can you please take this one?

I think in the main README we need Requirements / Prerequisites chapter with two subchapters - one for simple users and one for developers.

Users are interested in:
- Docker engine version - We use template 3.7
- docker-compose version

Developers are interested in:
- Go version
- protoc version - In order to build .proto files

This information should also be added in the offical docs with necessary instructions and troubleshooting you noted in the comment above.

Also, it would be great if you can bump Go and protoc versions in the CI script to the most recent that can work nicely together.

**Comment by nwneisen** [comment]
The protobuf versions were already updated in #902.  The CI is already using the latest version of Go (1.13) so I left it as is.
